### PR TITLE
[G2M] Incellbar custom styling

### PR DESCRIPTION
**Pull request by kc-leung**
branch off from https://github.com/EQWorks/lumen-table/pull/35

- Added scrollbar if no pagination
- Added formatValue option for each column data
- Added custom barColumnsColor for bar columns ( single color for now can be changed to multiple for different columns based on header key if needed )
- Added default filter search with title - HeaderTitle component

**Figma design file:** 
https://www.figma.com/file/Vr5v8g2MicUtWbcm3q7T3V/Paymi-Retail-Insights-Dashboard?node-id=304%3A105192&t=EBp78fTZPGMamyQQ-0

**Default with bar columns**

Screen Shot 2023-01-20 at 2 35 26 PM

**Custom styling with HeaderTitle component and formatValue resolution & marketshare columns**

Screen Shot 2023-01-20 at 2 39 47 PM

**Compact custom styling**

Screen Shot 2023-01-26 at 3 14 39 PM

### CODE REVIEW

File: src/in-cell-bar.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+const maxValsPerColumn = {}
+const computeMaxVals = (data, columnID, maxValsPerColumn) => {
+  if (!maxValsPerColumn[columnID]) {
+    const max = Math.max(...data.map((column) => parseInt(column[columnID])))
+    maxValsPerColumn[columnID] = max
+  }
+  return maxValsPerColumn
+}
+
+const adjustBarColor = (amount, color) => {
```
Comment (geoerika) [inline src/in-cell-bar.js:32]: I believe this is where you can use the d3 linear scale colour function?

File: src/index.js [hunk 2]
```diff
@@ -207,8 +241,23 @@ export const Table = forwardRef(({
     }
   }
 
+  const renderTableRow = data => (
+    data.map((row, i) => {
+      prepareRow(row)
+      return (
+        <tr className="table__body-row" key={i} {...row.getRowProps()}>
+          {row.cells.map((cell, i) => (
+            <td className={`table__body-cell border-${defaultStyles.borderType} border-secondary-200 text-secondary-800 ${i === (highlightColumn - 1) && 'font-bold'}`} key={i} {...cell.getCellProps()}>
```
Comment (geoerika) [inline src/index.js:250]: Could you please break down this long line on at least 2?

**Comment by geoerika** [comment]
@kc-leung this PR is still WIP.

**Review by geoerika — CHANGES_REQUESTED** [review]
Maybe we include the d3 functions to calculate colour hues & the extent of data. Let me know what you think.

**Comment by geoerika** [comment]
Ref for d3:
https://www.geeksforgeeks.org/d3-js-d3-extent-function/
https://github.com/EQWorks/react-maps/blob/main/src/components/locus-map/utils/scheme-color.js 
https://www.geeksforgeeks.org/d3-js-d3-extent-function/

File: stories/table-chart.stories.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        centerHeader: true,
+      }
+    }
+  />
+)
+
+export const customCompactStyle = () => (
+  <Table 
+    data={data} 
+    classes={{
```
Comment (geoerika) [inline stories/table-chart.stories.js:166]: I see that we will have to add this compact styling in widget studio when implementing the table there. How about we add a prop (ex: compact) and save these compact styling in lumen-table and add them in lumen-table if the prop is true?
This way we help users have a compact table without the hustle of thinking how to create this special styling to pass on to the table.

**Review by geoerika — CHANGES_REQUESTED** [review]
Just a note to think about and one too long of a line.

File: src/tableStyle.js [hunk 4]
```diff
@@ -1,109 +1,119 @@
 import { makeStyles } from '@eqworks/lumen-labs'
 
-const tableStyle = makeStyles(({
+const tableStyle = ({ headerTitle, centerHeader, compactTable }) => makeStyles(({
```
Comment (woozyking) [inline src/tableStyle.js:3]: Would recommend naming the module as `src/table-style.js` to stay consistent with the rest of the conventions

File: stories/data/dashboard-data.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
+const data = [
```
Comment (woozyking) [inline stories/data/dashboard-data.js:1]: `export default [...]` would do

**Review by geoerika — APPROVED** [review]
Other than Leo's comments, it looks good!

**Review by Hamaad102 — APPROVED** [review]

**Pull request merged**